### PR TITLE
Frontend/87 counterfactual panel

### LINKED ISSUE #87: Display number of days shift when counterfactual date is selected.
Currently we only have a placeholder for these, we need to calculate them in the Infopanel.

More shifts days into bottom card.

### PR DESCRIPTION
**Pull request by crangelsmith**
Fixes #87 

- Moving the restriction shift from CounterfactualStory to ConterfactualStatistics
- Doing the shift calculation as the difference of dates
- Made sure % reduction is only shown after the counterfactual date has changed

### CODE REVIEW

File: frontend/src/components/CounterfactualStatistics.jsx [hunk 1]
```diff
@@ -5,11 +5,15 @@ import PropTypes from "prop-types";
...
   totalCasesReal: PropTypes.number,
+  shiftFirstRestrictions: PropTypes.instanceOf(Date).isRequired,
+  shiftLockdown: PropTypes.instanceOf(Date).isRequired,
 });
 
 const defaultProps = {};
 
 const CounterfactualStatistics = (props) => {
+  // divide to turn shift from date difference to days
+  const to_days = 1000 * 3600 * 24;
```
Comment (jemrobinson) [inline frontend/src/components/CounterfactualStatistics.jsx]: Maybe this is better as a function? Something like:

```
function daysFromDate(inputDate) {
  return (inputDate / (1000 * 3600 * 24));
}
```

Also, can you explain why dividing converts a date into a number of days? I'm not sure I get that bit.

File: frontend/src/components/CounterfactualStatistics.jsx [hunk 2]
```diff
@@ -20,14 +24,21 @@ const CounterfactualStatistics = (props) => {
...
         {!props.totalCasesCounterfactual ? null : (
           <Card.Text>
             {`Total COVID-19 Cases per Million: ${props.totalCasesCounterfactual
               .toFixed(0)
               .toString()} \n `}
           </Card.Text>
         )}
-        {!(props.totalCasesCounterfactual & props.totalCasesReal) ? null : (
+        {!(props.shiftFirstRestrictions !== 0 || props.shiftLockdown !== 0) ||
+        !props.totalCasesCounterfactual ? null : (
```
Comment (jemrobinson) [inline frontend/src/components/CounterfactualStatistics.jsx]:
```suggestion
        !(props.totalCasesCounterfactual & props.totalCasesReal)? null : (
```

I think the check on `props.totalCasesReal` got accidentally dropped here?

File: frontend/src/components/CounterfactualStatistics.jsx [hunk 3]
```diff
@@ -20,14 +24,21 @@ const CounterfactualStatistics = (props) => {
...
+        } days`}</Card.Text>
         {!props.totalCasesCounterfactual ? null : (
           <Card.Text>
             {`Total COVID-19 Cases per Million: ${props.totalCasesCounterfactual
               .toFixed(0)
               .toString()} \n `}
           </Card.Text>
         )}
-        {!(props.totalCasesCounterfactual & props.totalCasesReal) ? null : (
+        {!(props.shiftFirstRestrictions !== 0 || props.shiftLockdown !== 0) ||
```
Comment (jemrobinson) [inline frontend/src/components/CounterfactualStatistics.jsx:46]: I think this should be a separate ternary check with the following contents:
```
<Card.Text>{`Shift first restrictions: ${props.shiftFirstRestrictions}`}</Card.Text>
<Card.Text>{`Shift lockdown: ${props.shiftLockdown}`}</Card.Text>
```

Or even two separate checks, one for each?

File: frontend/src/components/InfoPanel.jsx [hunk 4]
```diff
@@ -238,6 +236,14 @@ class InfoPanel extends React.Component {
                       this.state.totalCasesCounterfactual
                     }
                     totalCasesReal={this.state.totalCasesReal}
+                    shiftFirstRestrictions={
+                      new Date(this.state.dateFirstRestrictionsReal) -
+                      new Date(this.state.dateFirstRestrictionsCounterfactual)
+                    }
```
Comment (jemrobinson) [inline frontend/src/components/InfoPanel.jsx]: Should this be a member function that we call instead of something that happens in place?

**Review by jemrobinson — CHANGES_REQUESTED** [review]

File: frontend/src/components/CounterfactualStatistics.jsx (hunk 1 above)
Comment (crangelsmith) [reply frontend/src/components/CounterfactualStatistics.jsx]: Done. The date difference returns it in milliseconds, that is why we need to divide it.

File: frontend/src/components/InfoPanel.jsx (hunk 4 above)
Comment (crangelsmith) [reply frontend/src/components/InfoPanel.jsx]: Done.

File: frontend/src/components/CounterfactualStatistics.jsx (hunk 3 above)
Comment (crangelsmith) [reply frontend/src/components/CounterfactualStatistics.jsx:46]: Done.

**Comment by crangelsmith** [comment]
All comments have been resolved. I'll merge this branch now.

**Pull request merged**